Makefile

```diff
@@ -1,33 +1,48 @@
+NAMESPACE := $(shell python -c 'import yaml; print(yaml.safe_load(open("galaxy.yml"))["namespace"])')
+NAME := $(shell python -c 'import yaml; print(yaml.safe_load(open("galaxy.yml"))["name"])')
+VERSION := $(shell python -c 'import yaml; print(yaml.safe_load(open("galaxy.yml"))["version"])')
+MANIFEST := build/collections/ansible_collections/$(NAMESPACE)/$(NAME)/MANIFEST.json
+
+PLUGIN_TYPES := filter modules module_utils doc_fragments
+METADATA := galaxy.yml LICENSE README.md
+$(foreach PLUGIN_TYPE,$(PLUGIN_TYPES),$(eval _$(PLUGIN_TYPE) := $(wildcard plugins/$(PLUGIN_TYPE)/*.py)))
+DEPENDENCIES := $(METADATA) $(foreach PLUGIN_TYPE,$(PLUGIN_TYPES),$(_$(PLUGIN_TYPE)))
+
+PYTHON_VERSION = 3.7
 TEST=
-DATA=$(shell echo $(MODULE) | sed 's/\(foreman\|katello\)_//; s/_/-/g')
-PDB_PATH=$(shell find /usr/lib{,64} -type f -executable -name pdb.py 2> /dev/null)
-ifeq ($(PDB_PATH),)
-	PDB_PATH=$(shell which pdb)
-endif
-MODULE_PATH=plugins/modules/$(MODULE).py
-DEBUG_DATA_PATH=tests/debug_data/$(DATA).json
-DEBUG_OPTIONS=-m $(MODULE_PATH) -a @$(DEBUG_DATA_PATH) -D $(PDB_PATH)
-COLLECTION_TMP:=$(shell mktemp -du)
-COLLECTION_INSTALL_TMP:=$(shell mktemp -du)
 PYTEST=pytest -n 4 --boxed -v
 
 default: help
 help:
 	@echo "Please use \`make <target>' where <target> is one of:"
 	@echo "  help           to show this message"
-	@echo "  lint           to run flake8 and pylint"
+	@echo "  info           to show infos about the collection"
+	@echo "  lint           to run code linting"
 	@echo "  test           to run unit tests"
 	@echo "  sanity         to run santy tests"
-	@echo "  debug          debug a module using the ansible hacking module"
-	@echo "  setup          to set up test, lint, and debugging"
+	@echo "  setup          to set up test, lint"
 	@echo "  test-setup     to install test dependencies"
-	@echo "  debug-setup    to set up the ansible hacking module"
 	@echo "  test_<test>    to run a specific unittest"
 	@echo "  record_<test>  to (re-)record the server answers for a specific test"
-
-lint:
+	@echo "  clean_<test>   to run a specific test playbook with the teardown and cleanup tags"
+	@echo "  dist           to build the collection artifact"
+
+info:
+	@echo Building collection $(NAMESPACE)-$(NAME)-$(VERSION)
+	@echo   filter: $(notdir $(_filter))
```

Comment:
wonder if a loop over `PLUGIN_TYPES` would be cool or horrible here

Reply:
I had the same question. As a loop, this will end up almost unreadable methinks.

Reply:
ack

Reply:
Turns out to be not too bad. ;)